src/main/java/com/walmartlabs/x12/X12Parser.java

```diff
@@ -53,9 +64,28 @@
             return segments;
         } else {
             // if there is only one line in the source data
-            // we should attempt to use the segment separator passed in
-            // and see if we can split up this source data
-            return splitSourceDataIntoSegments(sourceData, "\\" + DEFAULT_SEGMENT_SEPARATOR);
+            // we should attempt to use the default segment separator
+            segments = splitSourceDataIntoSegments(sourceData, "\\" + DEFAULT_SEGMENT_SEPARATOR);
+            if (segments != null && segments.size() > 1) {
+                return segments;
+            } else {
+                // if there is only one line in the source data
+                // now try to get the specified delimiter
+                return splitSourceDataIntoSegments(sourceData, "\\" + findSegmentDelimiterCharacter(sourceData));
```

Comment:
can we create a String variable first
`String segmentSeparatorRegEx = "\\" + findSegmentDelimiterCharacter(sourceData))`
and then pass it to this call
`splitSourceDataIntoSegments (.. , ..)` easy to read